coalib/processes/Processing.py

```diff
@@ -102,7 +104,7 @@ def autoapply_actions(results,
                       file_dict,
                       file_diff_dict,
                       section,
-                      log_printer):
+                      log_printer=None):
```

Comment:
@nkprince007 
Exactly. That's what I'm trying to achieve. We want to remove the usage of `LogPrinter` objects (`log_printer` in most cases), but we can't remove it from the function parameter. Removing it from function parameters will lead to break in public API.
This PR is only trying deprecate it, and not remove it totally. After this, it will be easier remove it totally as it won't be used anywhere essentially. That PR will be done after this and be merged only when there's a major release of coala as it will be breaking many public APIs.